src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode, useState } from 'react';
+import Button from '@mui/material/Button';
+import ConfirmationDialog from './Confirmation/ConfirmationDialog';
+
+export type DialogProps = {

```

Comment:
If you split this into two pieces, you can reuse these everywhere - something like `DialogProps = { children: ReactNode; onCancel: () => void; sx?: { [key: string]: string | number }}` and `FormDialogProps = DialogProps & { onConfirm?: (formData: { [key: string]: string | number }) => void }`